TTKDTrainingApp/src/components/VideoWithControls/VideoWithControls.js

```diff
@@ -0,0 +1,162 @@
+import React, {useState, useRef, useEffect} from 'react';
+
+import {
+  StyleSheet,
+  View,
+  Dimensions,
+  TouchableWithoutFeedback,
+  ActivityIndicator,
+} from 'react-native';
+import {withNavigationFocus} from 'react-navigation';
+import {HeaderBackButton} from 'react-navigation-stack';
+
+import Video from 'react-native-video';
+
+import Controls from './Controls';
+import ContentVideo from './ContentVideo';
+
+import {DEFAULT_SPEED} from './constants';
+import Steps from './Steps';
+
+const VideoWithControls = props => {
+  /////////// Props/Navigation access////////////////
+  const contentId =
+    (props.navigation && props.navigation.getParam('contentId')) ||
+    props.contentId;
+  const recordedVideo =
+    props.navigation && props.navigation.getParam('recordedVideo');
+
+  //////////////// State ////////////////////////
+  const [paused, setPaused] = useState(true);
+  const [progress, setProgress] = useState(0);
+  const [duration, setDuration] = useState(0);
+  const [rate, setRate] = useState(DEFAULT_SPEED);
+  const [loading, setLoading] = useState(true);
+
+  ///////////////////// Effects /////////////////////
+
+  useEffect(() => {
+    if (!props.isFocused) {
```

Comment:
what is this isFocused prop ? Are we actually able to determine if a video is focused or not ?

Reply:
`isFocused` is provided by the `react-native-navigation` library. it signifies if that component is focused, aka (in this instance) if we're on that screen or not. If we aren't, then pause the video

https://reactnavigation.org/docs/en/with-navigation-focus.html